packages/application-shell/src/components/theme-switcher/theme-switcher.tsx

```diff
@@ -1,13 +1,36 @@
+import { useState } from 'react';
 import { useFeatureToggle } from '@flopflip/react-broadcast';
 import { themesOverrides } from '@commercetools-frontend/application-components';
 import { ThemeProvider } from '@commercetools-uikit/design-system';
+import { STORAGE_KEYS } from '../../constants';
 import { UI_REDESIGN } from '../../feature-toggles';
+import NewDesignReleaseInfoDialog from './new-design-release-info-dialog';
 
 const ThemeSwitcher = () => {
   const isNewThemeEnabled = useFeatureToggle(UI_REDESIGN);
   const theme = isNewThemeEnabled ? 'test' : 'default';
+  const [isOpen, setIsOpen] = useState(true);
+  const handleCloseDialog = () => {
+    window.localStorage.setItem(
+      STORAGE_KEYS.IS_NEW_DESIGN_RELEASE_NOTIFICATION_DIALOG,
+      'false'
+    );
+    setIsOpen(false);
+  };
+
   return (
-    <ThemeProvider theme={theme} themeOverrides={themesOverrides[theme]} />
+    <>
+      <ThemeProvider theme={theme} themeOverrides={themesOverrides[theme]} />
+      {isNewThemeEnabled &&
+      window.localStorage.getItem(
+        STORAGE_KEYS.IS_NEW_DESIGN_RELEASE_NOTIFICATION_DIALOG
+      ) === 'true' ? (
+        <NewDesignReleaseInfoDialog
+          onClose={handleCloseDialog}
+          isOpen={isOpen}
+        />
+      ) : null}
+    </>
```

Comment:
Perhaps we can just read the value from local storage at the beginning and conditionally render the dialog (I think we don't need the `isOpen` attribute).

What do you think is something like this?
```jsx
const ThemeSwitcher = () => {
  const isNewThemeEnabled = useFeatureToggle(UI_REDESIGN);
  const theme = isNewThemeEnabled ? 'test' : 'default';
  const [
    hasUserSeenNewDesignReleaseNotificationDialog,
    setHasUserSeenNewDesignReleaseNotificationDialog
  ] = useState(window.localStorage.getItem(
    STORAGE_KEYS.IS_NEW_DESIGN_RELEASE_NOTIFICATION_DIALOG
  ) === 'true');
  const handleCloseDialog = () => {
    window.localStorage.setItem(
      STORAGE_KEYS.IS_NEW_DESIGN_RELEASE_NOTIFICATION_DIALOG,
      'true'
    );
    setHasUserSeenNewDesignReleaseNotificationDialog(true)
  };

  return (
    <>
      <ThemeProvider theme={theme} themeOverrides={themesOverrides[theme]} />
      {!hasUserSeenNewDesignReleaseNotificationDialog ? (
        <NewDesignReleaseInfoDialog
          onClose={handleCloseDialog}
        />
      ) : null}
    </>
  );
};
```

Reply:
Yes this makes sense. On it.

Reply:
Looks like we will be needing the isOpen prop for the infoDialog component though.

Reply:
You just can pass `true` to the `isOpen` property of the `InfoDialog` component since it won't be rendered when we don't need it to be displayed.